### ISSUE TITLE
Implement recording of currents generated by electrodes in Brian

### ISSUE THREAD

**Issue by appukuttan-shailesh** [NeuralEnsemble/PyNN#465]
It would be useful to be able to record the currents generated by the electrodes. This is currently possible for the NEURON back-end via the _record() and _get_data() methods. These would be made public methods once the same are implemented for Brian and NEST.

The implementation of these methods for Brian could be as follows:
```python
    def _record(self):        
        self.i_state_monitor = brian.StateMonitor(self.cell_list[0].parent.brian_group[self.indices[0]], 'i_inj', record=0)
        simulator.state.network.add(self.i_state_monitor)

    def _get_data(self):
        return numpy.array((self.i_state_monitor.times / ms, self.i_state_monitor[0] / nA))
```

The NEST implementation is also more-or-less complete, but awaits the acceptance of the supporting features in the NEST simulator (see: https://github.com/nest/nest-simulator/issues/658). Once approved, the same would be posted separately, and the methods made public.

@apdavison : I split this into individual issues for Brian and NEST as the ability to record currents seemed necessary for testing the fix for issue #445 . Not trivial (and unnecessarily complex I presume) trying to do it without the current traces.

**Comment by appukuttan-shailesh** [comment]
Here is a sample output for each of the four electrode classes. The test file is [here](https://github.com/appukuttan-shailesh/PyNN_testData/blob/master/testCase_465_all.py).
Red: ACSource
Blue: DCSource
Green: NoisyCurrentSource
Black: StepCurrentSource
![brian_all](https://cloud.githubusercontent.com/assets/24866517/23958300/fb6ecbf4-09a1-11e7-95c5-afeb8009eb67.png)

**Comment by appukuttan-shailesh** [comment]
The recording of the currents can be tested more rigorously (and updated, if required) once implemented for other simulators (NEST remaining). As of now, one known difference between the Brian and NEURON implementation is that the data for the last time instant is not obtained from Brian. NEST would face the same situation (see [here](https://github.com/nest/nest-simulator/issues/658)). This can be regularized across simulator back-ends when incorporating this feature for NEST.

**Comment by apdavison** [comment]
Fixed in #467
